### DISCUSSION TITLE
How to create a primary key on an existing table without using `text()`?

### DISCUSSION THREAD

**Discussion by jrmidkiff** [sqlalchemy/sqlalchemy#11741] [Usage Questions]
I would like to create a possibly compound primary key on an already existing table in my postgresql database? Is that possible without using `text()`? I don't want to use that function because I can't guarantee that I am getting trusted input for the table name or the primary key column names

Thank you

**Comment by CaselIT** [comment]
Hi,

In general you will look at alembic for ddl migrations, but I'm not sure it suoports you use case. Creating a primary key given user input seems like a strange requirement

**Reply by jrmidkiff** [reply]
Thanks - looks like I'll need to get creative then

As to the why - we are functionally acting as database administrators who can alter other departments' tables based on the configuration information they give us
